preact/react-deps.js

```diff
@@ -14,6 +14,8 @@ export {
   useCallback,
 } from "preact/hooks";
 
+export { useSyncExternalStore } from "preact/compat";
```

Comment:
A bit concerned here that we're introducing a dependency here on preact/compact, I left a [comment regarding that](https://github.com/molefrog/wouter/pull/259#pullrequestreview-1161000871) when we were working on `forwardRef`. The implementation itself is simple and it'll be tree-shaken, but we're locking our preact version to `^10.11.3` which is only 4 days old. Perhaphs, it makes sense to bring the shim inside of the `preact/react-deps.js` file. We can drop it along with shim for React in future releases of course. What do you think?

Reply:
@molefrog sure, that's fine with me. I don't know enough about Preact to know if the React shim is compatible with Preact since it has a peer dependency on React, not Preact. (I don't use Preact at all so you'll have to tell me.) Is that the case?

If so, great.

If not, and if we wanted to use an older version of Preact, we could go back as far as Preact 10.9.0 (the first release to include it), however, that version does have a buggy implementation of `useSyncExternalStore`. The reason I put it at 10.11.3 was because I noticed that release included their latest bugfix for `useSyncExternalStore`. Why they didn't just copy React's shim implementation in the first place is beyond me.

Reply:
Gotcha. I would copy that latest version of the hook to the source code then. It's actually identical to what we had before, so I don't think we could introduce any regression.

Reply:
@molefrog done!